Source/WebCore/rendering/RenderReplaced.cpp

```diff
@@ -425,25 +425,35 @@ void RenderReplaced::computeAspectRatioInformationForRenderBox(RenderBox* conten
     }
+    constrainedSize = intrinsicSize;
+}
 
+void RenderReplaced::constrainIntrinsicSize(FloatSize& intrinsicSize, double& intrinsicRatio) const
+{
     // Now constrain the intrinsic size along each axis according to minimum and maximum width/heights along the
     // opposite axis. So for example a maximum width that shrinks our width will result in the height we compute here
     // having to shrink in order to preserve the aspect ratio. Because we compute these values independently along
     // each axis, the final returned size may in fact not preserve the aspect ratio.
-    // FIXME: In the long term, it might be better to just return this code more to the way it used to be before this
-    // function was added, since all it has done is make the code more unclear.
-    constrainedSize = intrinsicSize;
-    if (intrinsicRatio && !intrinsicSize.isEmpty() && style().logicalWidth().isAuto() && style().logicalHeight().isAuto()) {
+    if (intrinsicRatio && style().logicalWidth().isAuto() && style().logicalHeight().isAuto()) {
         if (isVideoWithDefaultObjectSize(this)) {
             LayoutUnit width = RenderBox::computeReplacedLogicalWidth();
-            constrainedSize.setWidth(width);
-            LayoutUnit height = blockSizeFromAspectRatio(horizontalBorderAndPaddingExtent(), verticalBorderAndPaddingExtent(), intrinsicRatio, BoxSizing::ContentBox, adjustBorderBoxLogicalWidthForBoxSizing(width, LengthType::Fixed));
-            constrainedSize.setHeight(height.round() - verticalBorderAndPaddingExtent());
+            intrinsicSize.setWidth(width);
+            LayoutUnit height = blockSizeFromAspectRatio(horizontalBorderAndPaddingExtent(), verticalBorderAndPaddingExtent(), intrinsicRatio, BoxSizing::ContentBox, adjustBorderBoxLogicalWidthForBoxSizing(width, LengthType::Fixed), style().aspectRatioType(), true);
+            intrinsicSize.setHeight(height.round() - verticalBorderAndPaddingExtent());
         } else {
-            // We can't multiply or divide by 'intrinsicRatio' here, it breaks tests, like fast/images/zoomed-img-size.html, which
-            // can only be fixed once subpixel precision is available for things like intrinsicWidth/Height - which include zoom!
-            constrainedSize.setWidth(RenderBox::computeReplacedLogicalHeight() * intrinsicSize.width() / intrinsicSize.height());
-            constrainedSize.setHeight(RenderBox::computeReplacedLogicalWidth() * intrinsicSize.height() / intrinsicSize.width());
+            auto removeBorderAndPaddingFromMinMaxSizes = [](LayoutUnit& minSize, LayoutUnit &maxSize, LayoutUnit borderAndPadding) {
+                minSize = std::max(0_lu, minSize - borderAndPadding);
+                maxSize = std::max(0_lu, maxSize - borderAndPadding);
+            };
+
+            auto [minLogicalWidth, maxLogicalWidth] = computeMinMaxLogicalWidthFromAspectRatio();
```

Comment:
Not sure if this is the right place to call `computeMinMaxLogicalWidthFromAspectRatio` which is calculating the transferred min/max width. I think logically, `computeMinMaxLogicalWidthFromAspectRatio` should be used in `RenderBox::computeReplacedLogicalWidthRespectingMinMaxWidth`, which is handling min/max size for replaced elements.
You may take a look at how non-replaced elements work, see `RenderBox::constrainLogicalMinMaxSizesByAspectRatio`.

Reply:
I moved this logic to `RenderBox::computeReplacedLogicalWidthRespectingMinMaxWidth`. I also modified `RenderReplaced::computeReplacedLogicalWidth` (and height) to no longer call these methods directly. Instead, these method are just providing a computed width/height, and then the result is being constrained by the caller so that it looks something like `computeReplacedLogicalWidthRespectingMinMaxHeight(computeReplacedLogicalWidth())`. There was already some code doing this in other parts of the code, too all of these calls should be consistent now. 

Once thing we could potentially do is just call `computeReplacedLogicalWidth` *within* `computeReplacedLogicalWidthRespectingMinMaxWidth,` so that it looks a little cleaner but let me know what you think.

Reply:
This change was causing a lot of issues so I reverted it back to the original version.